### ISSUE TITLE
heketi-cli hangs on deletion of device

### ISSUE THREAD

**Issue by t0ffel** [heketi/heketi#664]
1 disk in heketi glusterfs cluster failed. The disk was replaced with the new one. The OpenShift Enterprise cluster is freshly upgraded to 3.4.
I've delete volumes that used bricks on this device in heketi.
Now when I try to delete the device via heketi-cli it locks up:

[root@gprfc058 ~]# heketi-cli --version
heketi-cli 3.1.0
[root@gprfc058 ~]# heketi-cli -s http://heketi-storage-project.ose.sbu.lab.eng.bos.redhat.com device info 6be41332a98e512a56f5938a35351bbe
Device Id: 6be41332a98e512a56f5938a35351bbe
Name: /dev/sdc
State: online
Size (GiB): 279
Used (GiB): 0
Free (GiB): 279
Bricks:
[root@gprfc058 ~]# heketi-cli -s http://heketi-storage-project.ose.sbu.lab.eng.bos.redhat.com device delete 6be41332a98e512a56f5938a35351bbe

^^^ hangs here.

heketi pod image version:
registry.access.redhat.com/rhgs3/rhgs-volmanager-rhel7@sha256:2f01684b4cf1e78482ff8746e27e9f9fe44434479bafaeadb7c5dde9314ec6e6

heketi pod log:
....
[negroni] Started GET /queue/a038ffcd7862cdc68d3d248569bf0b6b
[negroni] Completed 200 OK in 35.812µs
[negroni] Started GET /devices/6be41332a98e512a56f5938a35351bbe
[negroni] Completed 200 OK in 305.341µs
[negroni] Started DELETE /devices/6be41332a98e512a56f5938a35351bbe
[heketi] INFO 2017/02/02 17:57:50 Deleting device 6be41332a98e512a56f5938a35351bbe on node ea266477606fb1f9e5a868e4801ec56a
[negroni] Completed 202 Accepted in 450.002µs
[asynchttp] INFO 2017/02/02 17:57:50 Started job 355134827b7ef32ed264b0f4a6f051cf
[negroni] Started GET /queue/355134827b7ef32ed264b0f4a6f051cf
[negroni] Completed 200 OK in 23.095µs
....

Please advise how I can remove the device manually(or some other way around this).

/cc @portante

**Comment by lpabon** [comment]
@t0ffel If the device is still in the db (you can do run a `topology info` command), then try deleting again.  If that goes for a while, then do the following:

1. Go to the node and try to understand why vgdelete and pvdelete are not working.
1. Try again.
1. If that does not work, then use `heketi-cli device disable <device id>`.  This will remove the device from the allocator and thus will not be used.

**Comment by t0ffel** [comment]
thanks Luis,

The device was still in topology info.
I've tried again via heketi-cli. `vgremove vg_6be41332a98e512a56f5938a35351bbe` was running on the box.
nothing happened though.. heketi-cli was still hangning.
I've ran `vgremove vg_6be41332a98e512a56f5938a35351bbe` manually on the box and it completed immediately.
```
[root@gprfc064 ~]# ps -ef|grep vgre
root      70664  70646  0 Feb02 ?        00:00:00 vgremove vg_6be41332a98e512a56f5938a35351bbe
root     129722  61095  0 19:59 pts/0    00:00:00 grep --color=auto vgre
[root@gprfc064 ~]# vgdisplay |grep 6be41332a
[root@gprfc064 ~]#
```
But the device is still in topology and I'm still unable to delete the device via heketi-cli - it hangs
